App/CourseApp/SharedUI/Animations/FlipEffect.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  FlipEffect.swift
+//  CourseApp
+//
+//  Created by Marcel Mravec on 21.05.2024.
+//
+
+import SwiftUI
+
+struct FlipEffect: GeometryEffect {
+    var animatableData: Double {
+        get { angle }
+        set { angle = newValue }
+    }
+    
+    @Binding var flipped: Bool
+    var angle: Double
+    let half = 0.5
+    let axis: (x: CGFloat, y: CGFloat)
+    let beginingOfEffect = 90.0
+    let endOfEffect = 270.0
+    
+    func effectValue(size: CGSize) -> ProjectionTransform {
+        // We schedule the change to be done after the view has finished drawing,
+        // otherwise, we would receive a runtime error, indicating we are changing
+        // the state while the view is being drawn.
+        DispatchQueue.main.async {
```

Comment:
isn't there better way how to solve this? Eg detect when the view did appeared and later run the animation?